### ISSUE TITLE
Leverage docker cache

### ISSUE THREAD

**Issue by 2m** [docker/build-push-action#7]
Would it possible to add support for using docker cache?

Especially for PR validation, it would be great if it would be possible to specify a docker image that would be used for resolving layer cache. That image name would be passed to a `docker build` option:

```
--cache-from strings      Images to consider as cache sources
```

Also the image should be pulled from the registry first.

**Comment by brunowowk** [comment]
Is there any workaround using just this action, or maybe using https://github.com/actions/cache?

**Comment by rickclare** [comment]
When (and if) the following PR#17 from docker/github-actions is merged, will this help with this request?

https://github.com/docker/github-actions/pull/17

**Comment by appleboy** [comment]
@2m @rickclare Already merged. You should close this issue.

**Comment by zappy-shu** [comment]
Note that though the backend image has been released I am still waiting on approval of the updated docs to release the action with the new input. Pushing to get this done ASAP.

Thank you for your contribution @appleboy

**Comment by bplower** [comment]
Until the action is updated, the cache-from arg can be used via the `INPUT_CACHE_FROMS` env var that was added. The [cache-from documentation](https://docs.docker.com/engine/reference/commandline/build/#specifying-external-cache-sources) indicates the cached image is only used if was built using the build arg `BUILDKIT_INLINE_CACHE=1`

```yaml
- name: build and push docker image
  uses: docker/build-push-action@v1
  env:
    INPUT_CACHE_FROMS: myregistry/myimage
  with:
    ...
    build_args: BUILDKIT_INLINE_CACHE=1
```

I'm pretty sure that should work, however I haven't had much luck with it. I'm using a multi-stage build, which the cache-from docs say requires cache manifests generated by `buildctl` rather than the resulting image, so I'm pretty sure my example above would work non-mulit-stage build.

I figured this was worth mentioning in case others are relatively new to the cache-from feature like myself 😄

**Comment by zappy-shu** [comment]
Merged and released. You can either use `docker/build-push-action@v1` or use `docker/build-push-action@v1.1.0` to use the new `cache_froms` input.

**Comment by robpc** [comment]
Can we have an example of how to use this option? Looking at the go code, it just adds the option to the build command, but that will do nothing when there is no local cache (ie a CB scenario like this). In my experience you need to pull the image first, so was hoping that this change did that as well. I did some testing with this action and was unable to get it to work which was consistent with that conclusion. 

```yaml
- name: Docker Build
  uses: docker/build-push-action@v1
  with:
    registry: quay.io
    repository: ${{ env.quayio_repo }}
    username: ${{ env.quayio_user }}
    password: ${{ secrets.QUAY_AUTH }}
    tags: >-
      ${{ env.docker_tag }},
      ${{ env.docker_cache_tag }}
    cache_froms: quay.io/${{ env.quayio_repo }}:${{ env.docker_cache_tag }}
```

**Comment by princemaple** [comment]
@robpc looks like you have to pull the cache_from image first.

**Comment by robpc** [comment]
@princemaple That was my conclusion, but my question is how to do that with this action.

**Comment by princemaple** [comment]
@robpc I used another one to login, run pull, then used this one to do the build. Did the job. Not sure if there is a better way though.

**Comment by robpc** [comment]
If you use another action to authenticate and pull then I am not sure why you wouldn't use that one to build and push as well. Since this comment hasn't gotten attention from the maintainers (understandable since the issue is closed), I will create a new issue requesting an example.
